core/src/test/java/io/atomix/core/TestMessagingService.java

```diff
@@ -76,19 +77,31 @@ public Address address() {
   }
 
   @Override
-  public CompletableFuture<Void> sendAsync(Address ep, String type, byte[] payload) {
-    return getHandler(ep, type).apply(address, payload).thenApply(v -> null);
+  public CompletableFuture<Void> sendAsync(Address address, String type, byte[] payload) {
+    return getHandler(address, type).apply(address, payload).thenApply(v -> null);
+  }
+
+  @Override
+  public CompletableFuture<byte[]> sendAndReceive(Address address, String type, byte[] payload) {
+    return getHandler(address, type).apply(address, payload);
+  }
+
+  @Override
+  public CompletableFuture<byte[]> sendAndReceive(Address address, String type, byte[] payload, Executor executor) {
+    ComposableFuture<byte[]> future = new ComposableFuture<>();
+    sendAndReceive(address, type, payload).whenCompleteAsync(future, executor);
+    return future;
   }
 
   @Override
-  public CompletableFuture<byte[]> sendAndReceive(Address ep, String type, byte[] payload) {
-    return getHandler(ep, type).apply(address, payload);
+  public CompletableFuture<byte[]> sendAndReceive(Address address, String type, byte[] payload, Duration timeout) {
+    return getHandler(address, type).apply(address, payload);
   }
 
   @Override
-  public CompletableFuture<byte[]> sendAndReceive(Address ep, String type, byte[] payload, Executor executor) {
+  public CompletableFuture<byte[]> sendAndReceive(Address address, String type, byte[] payload, Duration timeout, Executor executor) {
     ComposableFuture<byte[]> future = new ComposableFuture<>();
-    sendAndReceive(ep, type, payload).whenCompleteAsync(future, executor);
+    sendAndReceive(address, type, payload).whenCompleteAsync(future, executor);
```

Comment:
Shouldn't we be passing timeout into send and receive here?

Reply:
The `TestMessagingService` doesn't time out any messages ATM